src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -1607,10 +1606,31 @@ public async Task InboundWindowSize_Exceeded_Throw()
                 async uri =>
                 {
                     using HttpClient client = CreateHttpClient();
-                    using HttpResponseMessage response = await client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead);
 
-                    // Keep client open until server is done.
-                    await semaphore.WaitAsync(10000);
+                    Exception requestException = await Assert.ThrowsAsync<HttpRequestException>(async () =>
+                    {
+                        // GetAsync() tends to throw when running under high load.
+                        // This indicates enough content was buffered so a recv()
+                        // to pull in headers happened to read enough to exceed
+                        // the stream receive window.
+                        using HttpResponseMessage response = await client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead);
+
+                        // Keep client open until server is done.
+                        await semaphore.WaitAsync(10000);
+
+                        // ReadAsString() tends to throw under low load.
```

Comment:
Is there a bug with ReadAsString() we need to investigate? This comment doesn't really explain why we're coding up what looks like a "workaround" for a bug we need to fix elsewhere.

Reply:
I'll update the comments to better explain why an exception might come from either place.